x-pack/plugins/ml/public/application/app.tsx

```diff
@@ -50,11 +50,17 @@ export interface MlServicesContext {
 export type MlGlobalServices = ReturnType<typeof getMlGlobalServices>;
 
 const App: FC<AppProps> = ({ coreStart, deps, appMountParams }) => {
+  const redirectToMlAccessDeniedPage = async () => {
+    const accessDeniedPageUrl = `${coreStart.http.basePath.get()}/app/ml/access-denied`;
```

Comment:
I think this URL should be provided by the ML URL generator

Reply:
Updated here https://github.com/elastic/kibana/pull/76735/commits/cea397ee78ebf873893a6bc05185ddacaa40704c